ipld/read.go

```diff
@@ -42,22 +43,20 @@ func GetLeaves(ctx context.Context, dag ipld.NodeGetter, root cid.Cid, leaves []
 	return leaves, nil
 }
 
-// GetLeafData fetches and returns the data for leaf leafIndex of root rootCid.
-// It stops and returns an error if the provided context is canceled before
-// finishing
-func GetLeafData(
+// GetShare fetches and returns the data for leaf leafIndex of root rootCid.
+func GetShare(
```

Comment:
I got your point and agreed with it. However, the concept of Share [will leave outside](https://github.com/celestiaorg/celestia-node/pull/661#discussion_r861766526) of the node repo. For example, the definition of a share can leave in its repo, and all(rsmt2d, nmt, node) can rely on it. The `share` pkg in `node` will just rely on the `share` defined elsewhere and will only provide the logic around the share structures. 

The IPFS team did that to their low-level definition of data [Block](https://github.com/ipfs/go-block-format), and after years of this being a status quo, it turns out to be not that bad of an architecture decision. In our case, this can be like a globally defined Share format, which all other libs will depend on. Right now, we still have yet another Share definition in the Core/App and the nmt that we should eventually remove. Luckily we don't have multiple Share definitions anymore.

Reply:
I still like your idea of decoupling `ipld` from the notion of share. We can make this possible by extracting rsmt2d logic out of `ipld` pkg. The reason is that `rsmt2d` works with shares, not with IPLD leaves(ipld.Node) and this is the main reason for keeping Share within `ipld`. If we move rsmt2d-related  logic up to the `share`, then we can do what you suggest. We should make an issue about this.

Reply:
ack. This is really just a cleanliness/optimisation thing. I'll quickly make an issue.